tests/unit/modules/network/ios/test_ios_vrf_interfaces.py

```diff
@@ -0,0 +1,319 @@
+        )
+        self.get_config = self.mock_get_config.start()
+
+    def tearDown(self):
+        super(TestIosVrfInterfacesModule, self).tearDown()
+        self.get_resource_connection.stop()
+        self.get_config.stop()
+
+    def test_ios_vrf_interfaces_merged_idempotent(self):
+        self.maxDiff = None
+        run_cfg = dedent(
+            """\
+            interface GigabitEthernet1
+             ip address dhcp
+             no shutdown
+             !
+
+            interface GigabitEthernet2
+             no ip address
+             shutdown
+             !
+
+            interface GigabitEthernet3
+             no ip address
+             shutdown
+             !
+
+            interface GigabitEthernet4
+             no ip address
+             shutdown
+             !
+            """,
+        )
+        self.get_config.return_value = run_cfg
+        set_module_args(
+            dict(
+                config=[
+                    {"name": "GigabitEthernet1"},
+                    {"name": "GigabitEthernet2"},
+                    {"name": "GigabitEthernet3", "vrf_name": "testvrf2"},
+                    {"name": "GigabitEthernet3", "vrf_name": "testvrf2"},
+                    {"name": "GigabitEthernet4", "vrf_name": "testvrf1"},
+                ],
+                state="merged",
+            ),
+        )
+        commands = [
+            "interface GigabitEthernet4",
+            "vrf forwarding testvrf1",
+            "interface GigabitEthernet3",
+            "vrf forwarding testvrf2",
+        ]
+        result = self.execute_module(changed=True)
+        self.assertEqual(sorted(result["commands"]), sorted(commands))
+
+    def test_ios_vrf_interfaces_merged(self):
+        self.maxDiff = None
+        run_cfg = dedent(
+            """\
+            interface GigabitEthernet1
+             ip address dhcp
+             no shutdown
+             !
+
+            interface GigabitEthernet2
+             no ip address
+             shutdown
+             !
+
+            interface GigabitEthernet3
+             no ip address
+             shutdown
+             !
+
+            interface GigabitEthernet4
+             no ip address
+             shutdown
+             !
+            """,
+        )
+        self.get_config.return_value = run_cfg
+        set_module_args(
+            dict(
+                config=[
+                    {"name": "GigabitEthernet1"},
+                    {"name": "GigabitEthernet2"},
+                    {"name": "GigabitEthernet3"},
+                    {"name": "GigabitEthernet4", "vrf_name": "testvrf1"},
+                ],
+                state="merged",
+            ),
+        )
+        commands = [
+            "interface GigabitEthernet4",
+            "vrf forwarding testvrf1",
+        ]
+        result = self.execute_module(changed=True)
+        self.assertEqual(sorted(result["commands"]), sorted(commands))
+
+    def test_ios_vrf_interfaces_replaced(self):
```

Comment:
add idempotent checks for overridden and replaced 
also tests for gathered rendered parsed

Reply:
done.